From e5660338782c29e0ea9a0530d567ef27f6b4748d Mon Sep 17 00:00:00 2001
From: Brian Silverman <bsilverman@iveia.com>
Date: Wed, 11 Jun 2025 13:56:45 -0400
Subject: [PATCH] iveia version banner

Add iVeia version banner.

In addition, Xilinx banner includes a CRLF at the front, that often gets lost
on the console.  Add more to make start of boot more visible.

---
 lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c | 26 ++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
index a347d2f558..2aaa121d9a 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_main.c
@@ -36,6 +36,7 @@
 #include "xfsbl_hw.h"
 #include "xfsbl_main.h"
 #include "bspconfig.h"
+#include "iveia_version.h"
 
 /************************** Constant Definitions *****************************/
 
@@ -344,6 +345,7 @@ void XFsbl_PrintFsblBanner(void )
 	 */
 #if !defined(XFSBL_PERF) || defined(FSBL_DEBUG) || defined(FSBL_DEBUG_INFO) \
 			|| defined(FSBL_DEBUG_DETAILED)
+	XFsbl_Printf(DEBUG_PRINT_ALWAYS, "\r\n\r\n");
 	XFsbl_Printf(DEBUG_PRINT_ALWAYS,
                  "Zynq MP First Stage Boot Loader \n\r");
 	XFsbl_Printf(DEBUG_PRINT_ALWAYS,
@@ -353,6 +355,30 @@ void XFsbl_PrintFsblBanner(void )
 	XFsbl_Printf(DEBUG_GENERAL, "MultiBootOffset: 0x%0x\r\n",
 		XFsbl_In32(CSU_CSU_MULTI_BOOT));
 
+	XFsbl_Printf(DEBUG_PRINT_ALWAYS,"Machine:     iVeia %s\r\n", IVEIA_MACHINE);
+	XFsbl_Printf(DEBUG_PRINT_ALWAYS,"Src commit:  %s\r\n", IVEIA_SRC_BUILD_HASH);
+	#ifdef IVEIA_META_1_LAYER
+		XFsbl_Printf(DEBUG_PRINT_ALWAYS,"Layer (%s) commit: %s\r\n",
+		   IVEIA_META_1_LAYER, IVEIA_META_1_BUILD_HASH);
+	#else
+		XFsbl_Printf(DEBUG_PRINT_ALWAYS, "Layer commit: unknown\r\n");
+	#endif
+	#ifdef IVEIA_META_2_LAYER
+		XFsbl_Printf(DEBUG_PRINT_ALWAYS,"Layer (%s) commit: %s\r\n",
+		   IVEIA_META_2_LAYER, IVEIA_META_2_BUILD_HASH);
+	#endif
+	#ifdef IVEIA_META_3_LAYER
+		XFsbl_Printf(DEBUG_PRINT_ALWAYS,"Layer (%s) commit: %s\r\n",
+		   IVEIA_META_3_LAYER, IVEIA_META_3_BUILD_HASH);
+	#endif
+	#ifdef IVEIA_META_4_LAYER
+		XFsbl_Printf(DEBUG_PRINT_ALWAYS,"Layer (%s) commit: %s\r\n",
+		   IVEIA_META_4_LAYER, IVEIA_META_4_BUILD_HASH);
+	#endif
+	#if IVEIA_NUM_LAYERS > 4
+		#error More than four meta-iveia* layers not supported
+	#endif
+
 	if(FsblInstance.ResetReason == XFSBL_PS_ONLY_RESET) {
 		XFsbl_Printf(DEBUG_GENERAL,"Reset Mode	:	PS Only Reset\r\n");
 	} else if (XFSBL_MASTER_ONLY_RESET == FsblInstance.ResetReason) {
-- 
2.25.1

